Client/Impl/Worker/JobWorker.cs

```diff
@@ -104,14 +112,26 @@ internal void Open()
         var output = new ActionBlock<IJob>(activatedJob => { _ = Interlocked.Decrement(ref currentJobsActive); },
             executionOptions);
 
-        _ = input.LinkTo(transformer);
-        _ = transformer.LinkTo(output);
+        var linkInputTransformer = input.LinkTo(transformer);
+        var linkTransformerOutput = transformer.LinkTo(output);
 
         // Start polling
-        _ = Task.Run(async () => await PollJobs(input, cancellationToken),
-        cancellationToken).ContinueWith(
-        t => logger?.LogError(t.Exception, "Job polling failed."),
-        TaskContinuationOptions.OnlyOnFaulted);
+        pollingTask = Task.Run(async () => await PollJobs(input, cancellationToken),
+            cancellationToken).ContinueWith(
+        t =>
+        {
+            if (t.IsFaulted)
+            {
+                logger?.LogError(t.Exception, "Job polling failed");
+            }
+            else if (t.IsCanceled)
+            {
+                logger?.LogInformation("Job polling Cancelled");
+            }
+
+            linkInputTransformer.Dispose();
+            linkTransformerOutput.Dispose();
```

Comment:
❓ Should we maybe dispose the objects in our dispose? But I guess this will not work in the error case.